unitconfig/unitconfig.go

```diff
@@ -261,6 +289,10 @@ func (instance *Instance) handleNodeConfigStatus() {
 		if err := instance.client.SetNodeConfig(instance.unitConfig.Version, *nodeConfig); err != nil {
 			log.WithField("NodeID", nodeConfigStatus.NodeID).Errorf("Can't update node config: %v", err)
 		}
+
+		if nodeConfig.NodeID != nil && *nodeConfig.NodeID == instance.curNodeID {
```

Comment:
Not the right condition, let's discuss. Also should be called on update config.

Reply:
fixed